src/optimization/utils/constraints.jl

```diff
@@ -164,16 +171,18 @@ function encode_relu(::TriangularRelaxedLP, model, ẑᵢⱼ, zᵢⱼ, l̂ᵢⱼ
 end
 
 function encode_relu(::LinearRelaxedLP, model, ẑᵢⱼ, zᵢⱼ, δᵢⱼ)
-    @constraint(model, zᵢⱼ == (δᵢⱼ ? ẑᵢⱼ : 0.0))
+    @constraint(model, zᵢⱼ == (δᵢⱼ ? ẑᵢⱼ : 0.0)) # what does this mean? how can you use ? in a constraint?
```

Comment:
I agree, I am not sure what value δij will have during the creation of this constraint, or whether the @constraint macro is doing something unusual with that ternary operator.

Reply:
For the linear, triangular, and slack relaxations, `δᵢⱼ`  is not a variable of the model, but rather a concrete boolean value representing an actual (or potential) activation pattern. So in the case above, the expression could be, for example 
```julia
@constraint(model, zᵢⱼ == (true ? ẑᵢⱼ : 0.0))
# which evaluates to the following, because the expression is parsed inside-to-outside:
@constraint(model, zᵢⱼ == ẑᵢⱼ)
```
So for those two cases, there is no need (or capability) to constrain `δ`.

Reply:
gotchya ok good to know

Reply:
This comment too.